internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -1732,7 +1733,14 @@ func toProto(ctx context.Context, in target.Target, opt ...handlers.Option) (*pb
 		}
 	}
 	if outputFields.Has(globals.AddressField) {
-		out.Address = wrapperspb.String(in.GetAddress())
+		addr := in.GetAddress()
+		if addr != "" {
+			var err error
+			if addr, err = parseutil.NormalizeAddr(in.GetAddress()); err != nil {
+				return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("unable to normalize the given target address: %q", in.GetAddress()))
+			}
+		}
+		out.Address = wrapperspb.String(addr)
```

Comment:
This is odd -- Why do we need to normalize on output? Shouldn't it already be normalized from input?

Reply:
I don't think I noticed that it was only used for output addresses, just that toProto was called in every function. this may not be necessary as the tests were passing before I fixed the error return issue in 6f728b4891579eb7f438225f3031b9e0d309f962. for clarity, though, create and update were fixed by updating the `util.ParseAddress` function

Reply:
> create and update were fixed by updating the util.ParseAddress function

Yep, and we call that func in the Target repo so I just assumed that + changing the validation funcs (as you did) + changing any invalid tests should've done the job.